### PR TITLE
Differentiate entrypoints

### PR DESCRIPTION
**Pull request by eliamaino-fp**
See issue https://github.com/dailymotion/vast-client-js/issues/236

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,10 +22,12 @@
   "engines": {
     "node": "8.9.4"
   },
-  "main": "src/index.js",
+  "main": "vast-client-node.js",
```
Comment (jakub-g) [inline package.json]: This file is not generated in the repo, you forgot to `npm run build` and check it in :)

**Comment by jakub-g** [comment]
In general this seemed to be good, but I checked closely and the package size has exploded:

https://github.com/dailymotion/vast-client-js/blob/master/vast-client.js is 68 kb *unminified*
https://github.com/dailymotion/vast-client-js/blob/a953e068be2c58f080aeb02ade43aa8f68e2d615/vast-client.js is 153 kb *minified*

Then I did `$ webpack --json > stats.json` and uploaded to https://chrisbateman.github.io/webpack-visualizer/ that 72% of size are stuff from `node_modules` which come from node_url_handler and magically made their way into the bundle (check with http://webpack.github.io/analyse/)

Basically we need to get rid of node stuff from browser bundle

File: package.json (hunk 1 above)
Comment (eliamaino-fp) [reply package.json]: We don't commit the bundled versions, only for the release https://github.com/dailymotion/vast-client-js/blob/master/docs/CONTRIBUTING.md#pull-requests. We can definitely make an exception here :p

File: package.json [hunk 2]
```diff
@@ -22,10 +22,12 @@
...
+  "module": "src/index.js",
+  "browser": "vast-client.js",
```
Comment (jakub-g) [inline package.json]: Since we're changing this, maybe we should also rename those files as follows, in webpack build conf and package.json and in README / migration guide.

```
  "main": "dist/vast-client-node.js",
  "module": "src/index.js",
  "browser": "dist/vast-client-browser.js",
```

i.e. 1) move to `dist` and 2) optionally also append `-browser` in browser version. 
For people using automatic tools and `require('vast-client')`, this would non-breaking. This is breaking only for users who reference directly the filename. Anyway 2.0.0 is just out and probably no one is using it yet, so it's a good time to change if we want.

WDYT?

File: package.json (hunk 1 above)
Comment (jakub-g) [reply package.json]: ah fine 👍 let's do it before 2.0.1 ;)

File: package.json (hunk 2 above)
Comment (eliamaino-fp) [reply package.json]: Yes I agree with having a `dist` directory for the bundled files. And it should not be really considered a breaking change since the library is served as an `npm` package first. 👍

**Comment by eliamaino-fp** [comment]
After the splitting into multiple outputs (`dist` folder):

-  **Browser minified version**: `vast-client.min.js`, 
             -  48.5kb (total size)
             - 92.9% `src` / 7.1% `node_modules`
-  **Browser un-minified version**: `vast-client.js`, 
             -  132.3kb (total size)
             - 92.9% `src` / 7.1% `node_modules`
-  **Browser minified version**: `vast-client-node.min.js`, 
             -  71.0kb (total size)
             - 64.2% `src` / 35.7% `node_modules`
-  **Browser un-minified version**: `vast-client-node.js`, 
             -  195.4kb (total size)
             - 64.2% `src` / 35.7% `node_modules`

**Comment by eliamaino-fp** [comment]
**Browser builds:**
screen shot 2018-07-25 at 14 32 05

**Node builds:**
screen shot 2018-07-25 at 14 32 20

File: src/urlhandlers/mock_node_url_handler.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+export class NodeURLHandler {
```
Comment (jakub-g) [inline src/urlhandlers/mock_node_url_handler.js]: I'd add a remark in `url_handler` above this line

     import { XHRURLHandler } from './urlhandlers/xhr_url_handler';

telling that for a browser build, this will be replaced to a mock at build-time, so that the people reading the code will understand it

File: webpack.config.js [hunk 4]
```diff
@@ -1,29 +1,54 @@
...
+  } else {
+    // Options for browser bundling
+    config.resolve.alias['./urlhandlers/node_url_handler'] =
+      './urlhandlers/mock_node_url_handler';
+    config.output.filename =
+      env === 'dev' ? 'vast-client.js' : 'vast-client.min.js';
+    config.output.library = 'VAST';
+    config.node = {
+      fs: 'empty'
+    };
```
Comment (jakub-g) [inline webpack.config.js]: the last 3 lines are _probably_ not needed anymore due to the mocking that was put in place
